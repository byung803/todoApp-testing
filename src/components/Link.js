import React from 'react';

export const Link = ({ children, active, onClick }) => (
    <button
        onClick={onClick}
        disabled={active}
    >
        {children}</button>
)

export default Link;