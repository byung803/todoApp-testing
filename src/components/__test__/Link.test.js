import React from 'react';
import { shallow } from 'enzyme';
import { Link } from '../Link';

let wrapper, onClickSpy;

beforeEach(() => {
    onClickSpy = jest.fn();
    wrapper = shallow(<Link onClick={onClickSpy} disabled={true} children={[]} />);
})

describe('Link Component', () => {
    it('should have a button', () => {
        expect(wrapper.find('button').length).toBe(1);
    })

    it('should execute onClick method when clicked', () => {
        wrapper.find('button').simulate('click')
        expect(onClickSpy).toHaveBeenCalled();
    })

    it('should have snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })
})