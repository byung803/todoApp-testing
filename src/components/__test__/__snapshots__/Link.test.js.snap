// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Component should have snapshot 1`] = `
<button
  onClick={[MockFunction]}
/>
`;
